### PR TITLE
[ENG-1453] Add header test

### PR DESCRIPTION
**Pull request by adlius**
- Ticket: [ENG-1453]
- Feature flag: n/a

## Purpose

Add header test to check whether the header is set.

## QA Notes

No QA needed.

[ENG-1453]: https://openscience.atlassian.net/browse/ENG-1453

### CODE REVIEW

File: tests/integration/services/current-user-test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        const acceptHeaderExpected = `application/vnd.api+json; version=${apiVersion}`;
+        server.get('/users', (_: any, request: Request & { requestHeaders: { Accept: string } }) => {
+            assert.equal(request.requestHeaders.Accept, acceptHeaderExpected);
+            return fakeResponse;
+        });
+        server.post('/users', (_: any, request: Request & { requestHeaders: { Accept: string } }) => {
+            assert.equal(request.requestHeaders.Accept, acceptHeaderExpected);
+            return fakeResponse;
+        });
+        this.store.findAll('user');
```
Comment (jamescdavis) [inline tests/integration/services/current-user-test.ts]:
```suggestion
        await this.store.findAll('user');
```

File: tests/integration/services/current-user-test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        this.store.createRecord('user').save();
```
Comment (jamescdavis) [inline tests/integration/services/current-user-test.ts]:
```suggestion
        await this.store.createRecord('user').save();
```

File: tests/integration/services/current-user-test.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        assert.expect(2);
```
Comment (jamescdavis) [inline tests/integration/services/current-user-test.ts]: Normally, we'd put this at the beginning of the test.

**Comment by jamescdavis** [comment]
It's also be good to have a test call `authenticatedAjax` on the `currentUser` service and make sure the API version header is being passed there as well.

**Comment by adlius** [comment]
@jamescdavis Resolved.

**Pull request merged**